core/src/main/java/org/apache/accumulo/core/client/impl/TableOperationsImpl.java

```diff
@@ -1000,17 +1032,21 @@ public void setLocalityGroups(String tableName, Map<String,Set<Text>> groups)
             "Group " + entry.getKey() + " overlaps with another group");
       }
 
+      if(entry.getValue().isEmpty()) {
+        throw new IllegalArgumentException("Group "+entry.getKey()+" is empty");
+      }
+
       all.addAll(entry.getValue());
     }
 
     for (Entry<String,Set<Text>> entry : groups.entrySet()) {
       Set<Text> colFams = entry.getValue();
       String value = LocalityGroupUtil.encodeColumnFamilies(colFams);
-      setProperty(tableName, Property.TABLE_LOCALITY_GROUP_PREFIX + entry.getKey(), value);
+      setPropertyNoChecks(tableName, Property.TABLE_LOCALITY_GROUP_PREFIX + entry.getKey(), value);
```

Comment:
Wouldn't you also want to check here?

Reply:
This code is setting properties in the correct order and I did not want it to read all properties for each property set by this code.